### PR TITLE
Update outdated comments

### PR DESCRIPTION
**Pull request by Disservin**
Also add a documentative link about the fast modulo reduction

No functional change

### CODE REVIEW

**Comment by Disservin** [comment]
any other outdated comments while we're at it?

**Comment by gahtan-syarif** [comment]
@Disservin might want to update the elo on recapture extensions LTC https://tests.stockfishchess.org/tests/live_elo/660d3325216a13d9498e886a

File: src/search.cpp
Comment (Vizvezdenec) [inline src/search.cpp:1]: I would phrase it precisely tbh, so
"Extension for capturing the piece moved at the previous move"

File: src/misc.h [hunk 1]
```diff
@@ -182,6 +182,7 @@ class PRNG {
 
 inline uint64_t mul_hi64(uint64_t a, uint64_t b) {
 #if defined(__GNUC__) && defined(IS_64BIT)
+    // https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction/
```
Comment (mstembera) [inline src/misc.h:185]: I don't think this is necessary.

**Pull request closed without merging**